server/src/main/java/com/ncc/neon/services/ClusterService.java

```diff
@@ -31,25 +44,11 @@
      * @return the new clustered results
      */
     public TabularQueryResult cluster(TabularQueryResult tabularQueryResult) {
-        try {
-            if (this.clusterClause.getFieldType().equals("number")) {
-                return this.aggregateNumber(tabularQueryResult);
-            }
-        } catch (Exception e) {
-            e.printStackTrace();
-        }
+        String fieldType = this.clusterClause.getFieldType();
 
-        this.clusterClause = null; // reset cluster clause
-        return tabularQueryResult;
-    }
+        // to keep the check simple, specialty cases are for text
+        boolean isText = (fieldType.equals("text") || fieldType.equals("keyword")) ? true : false;
```

Comment:
Thinking forward: You may want to change isText from a boolean to an Enum.  That may make it easier to handle dates and lat/lon points in the future tickets.

Reply:
I understand making fieldType into the FieldType enum, but I'm not following what to make isText into. Maybe I should just use the FieldType enum and do another compare?

Reply:
Currently this method handles two use cases: `isText=true` and `isText=false`.  If this method will ever need to handle more than two use cases in the future (like dates and lat/lon points), then you may want to use an Enum.  Just a suggestion.  You can worry about it when it becomes relevant, if that's easier.

Reply:
I'm changing this on the date one.